src/Patches/ItemPatches.cs

```diff
@@ -372,7 +372,9 @@ public static ItemResult GiveItem(string ItemName, ItemInfo itemInfo) {
                 if (TunicRandomizer.Settings.BonusStatUpgradesEnabled) {
                     GoldenTrophy.collectionMessage = ScriptableObject.CreateInstance<LanguageLine>();
                     GoldenTrophy.collectionMessage.text = Translations.TranslateDefaultNoQuotes(ItemLookup.BonusUpgrades[Item.ItemNameForInventory].CustomPickupMessage, true);
-                    Inventory.GetItemByName(ItemLookup.BonusUpgrades[Item.ItemNameForInventory].LevelUp).Quantity += 1;
+                    string bonus_upgrade = ItemLookup.BonusUpgrades[Item.ItemNameForInventory].LevelUp;
+                    Inventory.GetItemByName(bonus_upgrade).Quantity += 1;
+                    SaveFile.SetInt(bonus_upgrade, SaveFile.GetInt($"randomizer bonus upgrade {bonus_upgrade}") + 1);
```

Comment:
```suggestion
                    string saveFlag = $"randomizer bonus upgrade {bonus_upgrade}";
                    SaveFile.SetInt(saveFlag, SaveFile.GetInt(saveFlag) + 1);
```
SetInt still isn't using the right flag in both cases